### PR TITLE
DEPR: make_block

### LINKED ISSUE #56815: DEPR: make_block
`make_block` was deprecated and then reverted before the 2.2 release: https://github.com/pandas-dev/pandas/pull/56481

On the dev call today (2024-01-10), it was agreed that a deprecation would be issued during 3.0

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #56815 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]
LGTM cc @jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
I have been working on a branch for pyarrow testing an alternative `to_pandas` implementation. I will try to finish that this week, and then I will be able to give more concrete feedback here.

**Comment by jorisvandenbossche** [comment]
BTW I posted some comments on the issue last week: https://github.com/pandas-dev/pandas/issues/56815

**Comment by jbrockmendel** [comment]
@jorisvandenbossche with #58197 in are we good to move forward here?

**Comment by jbrockmendel** [comment]
If there are no comments I plan to merge next week

File: pandas/core/internals/api.py [hunk 1]
```diff
@@ -87,6 +89,14 @@ def make_block(
     - Block.make_block_same_class
     - Block.__init__
     """
+    warnings.warn(
+        # GH#56815
+        "make_block is deprecated and will be removed in a future version. "
+        "Use public APIs instead.",
+        DeprecationWarning,
+        stacklevel=find_stack_level(),
```
Comment (jorisvandenbossche) [inline pandas/core/internals/api.py:97]:
```suggestion
        2,
```

File: pandas/core/internals/api.py [hunk 2]
```diff
@@ -87,6 +89,14 @@ def make_block(
     - Block.make_block_same_class
     - Block.__init__
     """
+    warnings.warn(
+        # GH#56815
+        "make_block is deprecated and will be removed in a future version. "
+        "Use public APIs instead.",
```
Comment (jorisvandenbossche) [inline pandas/core/internals/api.py:95]: Do we want to point to the new `pd.api.internals.create_dataframe_from_blocks`? I know we don't want to encourage that, but TBH I assume this is used by few libraries for a specific reason anyway (exactly the use case for which the pd.api.internals can be used for), and the docstring of that function also asks the user to raise an issue about it

File: pandas/io/feather_format.py [hunk 3]
```diff
@@ -123,9 +124,16 @@ def read_feather(
...
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore",
+                    "make_block is deprecated",
+                    DeprecationWarning,
+                )
+
+                return feather.read_feather(
+                    handles.handle, columns=columns, use_threads=bool(use_threads)
+                )
```
Comment (jorisvandenbossche) [inline pandas/io/feather_format.py:136]: Do we still need this? (the fix is in a released pyarrow, so with latest of both you won't see any warning. Although of course when testing with an older pyarrow you will see it, but I am not sure we added such warning suppressions for other deprecations like DataFrame(mgr)?)

**Review by jorisvandenbossche — COMMENTED** [review]
@jbrockmendel apologies for the very slow follow-up here! 
Added a few minor questions, but yes, in general I am fine with moving forward with this now (pyarrow 16.0 is released with the change for this, so that's the main thing I would have wanted to wait on)

File: pandas/io/feather_format.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/io/feather_format.py:136]: we dont _need_ it, no.  we need to catch this either here or filterwarnings in a bunch of test places.  so i prefer to catch it one place, but OK either way

**Comment by jbrockmendel** [comment]
Updated, I think addressing all comments

**Review by jorisvandenbossche — APPROVED** [review]
Thanks for the update! Looks good

**Pull request merged**